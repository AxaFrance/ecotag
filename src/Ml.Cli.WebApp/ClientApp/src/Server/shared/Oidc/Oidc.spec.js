import React from "react";
import '@testing-library/jest-dom';
import {render} from '@testing-library/react';
import AuthenticatingError from "./AuthenticateError.component";
import Authenticating from "./Authenticating.component";
import {CallBackSuccess} from "./Callback.component";
import Loading from "./Loading.component";
import ServiceWorkerNotSupported from "./ServiceWorkerNotSupported.component";
import SessionLost from "./SessionLost.component";

describe('Oidc', () => {
   /* test('Render AuthenticatingError component', async () => {
        const { asFragment } = render(<AuthenticatingError />);
        expect(asFragment()).toMatchSnapshot();
    });*/

    test('Render Authenticating component', async () => {
        const { asFragment } = render(<Authenticating />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Render CallBackSuccess component', async () => {
        const { asFragment } = render(<CallBackSuccess />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Render Loading component', async () => {
        const { asFragment } = render(<Loading />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Render ServiceWorkerNotSupported component', async () => {
        const { asFragment } = render(<ServiceWorkerNotSupported />);
        expect(asFragment()).toMatchSnapshot();
    });

    /*test('Render SessionLost component', async () => {
        const { asFragment } = render(<SessionLost />);
        expect(asFragment()).toMatchSnapshot();
    });*/
});
