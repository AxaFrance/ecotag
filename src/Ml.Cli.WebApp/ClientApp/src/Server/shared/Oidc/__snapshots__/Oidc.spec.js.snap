// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Oidc Render Authenticating component 1`] = `
<DocumentFragment>
  <div
    class="oidc-authenticating"
    style="color: rgb(53, 110, 255); position: absolute; z-index: 1000; top: 0px; bottom: 0px; right: 0px; left: 0px;"
  >
    <div
      class="oidc-authenticating__container"
    >
      <h1
        class="oidc-authenticating__title"
      >
        Authentification en cours
      </h1>
      <p
        class="oidc-authenticating__content"
      >
        Vous serez redirigé vers la page de connexion.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Oidc Render AuthenticatingError component 1`] = `
<DocumentFragment>
  <div
    class="oidc-authenticating"
    style="color: rgb(53, 110, 255); position: absolute; z-index: 1000; top: 0px; bottom: 0px; right: 0px; left: 0px;"
  >
    <div
      class="oidc-authenticating__container"
    >
      <h1
        class="oidc-authenticating__title"
      >
        Authentification erreur
      </h1>
      <p
        class="oidc-authenticating__content"
      >
        Une erreur s'est produite lors de l'authentification.
      </p>
      <button
        class="btn af-btn"
        name="reauthenticate"
      >
        <span
          class="af-btn-text"
        >
          Ré-authentifier
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Oidc Render CallBackSuccess component 1`] = `
<DocumentFragment>
  <div
    class="oidc-callback"
    style="color: rgb(53, 110, 255); position: absolute; z-index: 1000; top: 0px; bottom: 0px; right: 0px; left: 0px;"
  >
    <div
      class="oidc-callback__container"
    >
      <h1
        class="oidc-callback__title"
      >
        Authentification terminée
      </h1>
      <p
        class="oidc-callback__content"
      >
        Vous allez être redirigé vers votre application.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Oidc Render Loading component 1`] = `
<DocumentFragment>
  <span
    class="oidc-loading"
    style="color: rgb(53, 110, 255); position: absolute; z-index: 1000; top: 0px; bottom: 0px; right: 0px; left: 0px;"
  >
    Chargement du module d'authentification
  </span>
</DocumentFragment>
`;

exports[`Oidc Render ServiceWorkerNotSupported component 1`] = `
<DocumentFragment>
  <div
    class="oidc-serviceworker"
    style="color: rgb(53, 110, 255); position: absolute; z-index: 1000; top: 0px; bottom: 0px; right: 0px; left: 0px;"
  >
    <div
      class="oidc-serviceworker__container"
    >
      <h1
        class="oidc-serviceworker__title"
      >
        Impossible de s'authentifier sur ce navigateur
      </h1>
      <p
        class="oidc-serviceworker__content"
      >
        Votre navigateur n’est pas assez sécurisé pour que l’authentification fonctionne. Essayez de mettre à jour votre navigateur ou d’utiliser un navigateur plus récent.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Oidc Render SessionLost component 1`] = `
<DocumentFragment>
  <div
    class="oidc-session-lost"
    style="color: rgb(53, 110, 255); position: absolute; z-index: 1000; top: 0px; bottom: 0px; right: 0px; left: 0px;"
  >
    <div
      class="oidc-session-lost__container"
    >
      <h1
        class="oidc-session-lost__title"
      >
        Session expirée
      </h1>
      <p
        class="oidc-session-lost__content"
      >
        Votre session a expiré. Veuillez vous ré-authentifier.
      </p>
      <button
        class="btn af-btn"
        name="reauthenticate"
      >
        <span
          class="af-btn-text"
        >
          Ré-authentifier
        </span>
      </button>
    </div>
  </div>
  <div />
</DocumentFragment>
`;
